### ISSUE TITLE
Logout issue: Required String parameter 'redirect_uri' is not present

### ISSUE THREAD

**Issue by eva-mueller-coremedia** [jenkinsci/oic-auth-plugin#229]
### Jenkins and plugins versions report

Plugin: `oic-auth-plugin`:2.5

### What Operating System are you using (both controller, and any agents involved in the problem)?

Jenkins 2.387.3

### Reproduction steps

* Install & configure `oic-auth-plugin` plugin in version 2.5
* Login to Jenkins via openid connect server
* Logout
* Error message is shown

### Expected Results

After logout, the user will be redirected to the login page.

### Actual Results

Error message

![image](https://github.com/jenkinsci/oic-auth-plugin/assets/14927927/19a0490c-a75e-4ed3-bdae-d3616d15f060)

### Anything else?

See also https://issues.jenkins.io/browse/JENKINS-66459

**Comment by eva-mueller-coremedia** [comment]
@michael-doubez Hi, if I can help with more information to get this issue solved, please feel free to reach out. It would be great to see this pop up vanish 🪄 . 
Thank you so far.

**Comment by michael-doubez** [comment]
@eva-mueller-coremedia from the Jira ticket, it looks like you are trying to fit parameters in the end session URL.

Normally, the redirect is provided through the post_logout_url parameter.

The current implmentation assumes the enpoint doesn't contains parameters:

https://github.com/jenkinsci/oic-auth-plugin/blob/49c7f67345c4144d693660453748763391ba0de1/src/main/java/org/jenkinsci/plugins/oic/OicSecurityRealm.java#L1006

**Comment by eva-mueller-coremedia** [comment]
> @eva-mueller-coremedia from the Jira ticket, it looks like you are trying to fit parameters in the end session URL.

Yes, that's right. I set the following for the **End session URL for OpenID Provider**:
`https://***.amazoncognito.com/logout?client_id=***&logout_uri=JENKINS_URL/logout`
by following the example: https://itnext.io/jenkins-saml-authentication-via-okta-and-users-groups-476a5763257c

Unfortunately, the AWS Cognito IdP’s Discovery Endpoint https://cognito-idp.REGION.amazonaws.com/REGION_POOLID/.well-known/openid-configuration does not return an `end_session_endpoint` URL entry, i.e. AWS Cognito does not follow the OpenID Connect RP-Initiated Logout specification 
... as you already pointed out.

<details>
<summary>Discovery Endpoint Json</summary>

```
{
	"authorization_endpoint": "https://***.amazoncognito.com/oauth2/authorize",
	"id_token_signing_alg_values_supported": [
		"RS256"
	],
	"issuer": "https://cognito-idp.REGION.amazonaws.com/REGION_POOLID",
	"jwks_uri": "https://cognito-idp.REGION.amazonaws.com/REGION_POOLID/.well-known/jwks.json",
	"response_types_supported": [
		[...]
	],
	"scopes_supported": [
		[...]
	],
	"subject_types_supported": [
		[...]
	],
	"token_endpoint": "https://***.amazoncognito.com/oauth2/token",
	"token_endpoint_auth_methods_supported": [
		[...]
	],
	"userinfo_endpoint": "https://***.amazoncognito.com/oauth2/userInfo"
}
```
</details>

The AWS Cognito [logout-endpoint](https://docs.aws.amazon.com/cognito/latest/developerguide/logout-endpoint.html) expects two request parameters (`client_id`, `logout_uri` or `redirect_uri `) and only supports `HTTPS GET` requests.

Nevertheless, it would be great to get support for the AWS Cognito logout. 🙇‍♀️

**Comment by michael-doubez** [comment]
@eva-mueller-coremedia note that you don't *really* need to logout from the provider (cognito in your case)

The most important is that session on the Jenkins side is terminated.
Logging out from the provider only invalidates the token which very likely is already expired or nearly so (~5min is the norm).

I would advise to simply not configure it unless you have a compeling reason to do so.

**Comment by eva-mueller-coremedia** [comment]
Hi @michael-doubez - thank you for your quick reply.

Im currently running **Jenkins** 2.401.1 and **oic-auth** 2.6.

### Use Case 1: **Configure** Logout from OpenID Provider

When I have set option set to `https://***.amazoncognito.com/logout?client_id=***&logout_uri=JENKINS_URL/logout`
and click the Jenkins Logout button, I am redirected to the Cognito login UI (after I dismissed the error message "Required string parameter...."). The user _seems to be_ logged out.

**I am wondering if the (Jenkins) user session is really terminated?** 
Do you know how to verify this? Didn't find a simple way...

### Use Case 2: **NOT Configure** Logout from OpenID Provider

When I omit the option and hit the Jenkins logout button, then I see the following page  ⚠️ **but the user is still logged in** ⚠️

<img width="695" alt="oic-logout" src="https://github.com/jenkinsci/oic-auth-plugin/assets/14927927/05a5ec3a-2e99-4414-a5d6-ceb4bdfd7c78">

The logout redirect me to: `JENKINS_URL/OicLogout/`

### Question

> Logging out from the provider only invalidates the token which very likely is already expired or nearly so (~5min is the norm).

Which token do you refer to? Within Cognito, you can customize the token expiration, e.g., 
* Refresh token expiration: 30 day(s)
* Access token expiration: 1 day(s)
* ID token expiration: 1 day(s)

**Comment by eva-mueller-coremedia** [comment]
@michael-doubez I opened a PR #241 which tackles this topic.

**Comment by eva-mueller-coremedia** [comment]
> @eva-mueller-coremedia note that you don't _really_ need to logout from the provider (cognito in your case)
> 
> The most important is that session on the Jenkins side is terminated. Logging out from the provider only invalidates the token which very likely is already expired or nearly so (~5min is the norm).
> 
> I would advise to simply not configure it unless you have a compeling reason to do so.

NOT Configure Logout from OpenID Provider does not work:
https://github.com/jenkinsci/oic-auth-plugin/issues/229#issuecomment-1586346218

@michael-doubez Should I open another issue?
